### PR TITLE
Fix app path setting during engine initialization

### PR DESCRIPTION
**Pull request by varyonic**
I offer this alternative to #5076 that does not rely on further changes to the Settings mixin, in particular read_default_setting().

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,10 +8,6 @@ dist: trusty
 
 bundler_args: --without development
 
-branches:
-  only:
-    - master
-
```
Comment (javierjulio) [inline .travis.yml]: Hmm that's weird, how come it was always running builds on PR's then?

Comment (deivid-rodriguez) [inline .travis.yml]: @varyonic sometimes likes to run CI on a branch without opening a PR. With our current Travis CI settings, this is the only way to do it.

Comment (javierjulio) [inline .travis.yml]: In this case running travis on their own fork correct?

Comment (varyonic) [inline .travis.yml]: Yes.  Deleted this commit.

**Comment by varyonic** [comment]
I looked at the CodeClimate issue and I'm not planning to address it.

**Comment by javierjulio** [comment]
@varyonic @Fivell I don't want to hold this up but I was thinking about this over the weekend. I realized the question didn't come up about what is even for? I realized it seemed weird to have to specify the app path. Is it really necessary?

**Comment by varyonic** [comment]
The idea is that app_path is valid even before Rails.root is set during initialization.  It's used to set load_path that in turn is used in #files to load the resource DSL.  Rails.root is only unset in certain corner cases, viz. application_spec.rb.  It might be appropriate, however, to update the [documented use of load_paths](https://github.com/activeadmin/activeadmin/blob/22da13f5ff330b1c89a19c026ee40432890e072f/docs/1-general-configuration.md)

**Pull request merged**